flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -178,77 +211,14 @@ class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
     }
   }
 
-  Future<void> _startTransaction(Route<dynamic>? route) async {
-    if (!_enableAutoTransactions) {
-      return;
-    }
+  Future<void> _startTimeToDisplayTracking(Route<dynamic>? route) async {
+    _completedDisplayTracking = Completer<void>();
+    final routeName = _getRouteName(route);
+    _currentRouteName = routeName;
 
-    String? name = _getRouteName(route);
     final arguments = route?.settings.arguments;
-
-    if (name == null) {
-      return;
-    }
-
-    if (name == '/') {
-      name = 'root ("/")';
-    }
-    final transactionContext = SentryTransactionContext(
-      name,
-      'navigation',
-      transactionNameSource: SentryTransactionNameSource.component,
-      // ignore: invalid_use_of_internal_member
-      origin: SentryTraceOrigins.autoNavigationRouteObserver,
-    );
-
-    _transaction = _hub.startTransactionWithContext(
-      transactionContext,
-      waitForChildren: true,
-      autoFinishAfter: _autoFinishAfter,
-      trimEnd: true,
-      onFinish: (transaction) async {
-        _transaction = null;
-        final nativeFrames = await _native
-            ?.endNativeFramesCollection(transaction.context.traceId);
-        if (nativeFrames != null) {
-          final measurements = nativeFrames.toMeasurements();
-          for (final item in measurements.entries) {
-            final measurement = item.value;
-            transaction.setMeasurement(
-              item.key,
-              measurement.value,
-              unit: measurement.unit,
-            );
-          }
-        }
-      },
-    );
-
-    // if _enableAutoTransactions is enabled but there's no traces sample rate
-    if (_transaction is NoOpSentrySpan) {
-      _transaction = null;
-      return;
-    }
-
-    if (arguments != null) {
-      _transaction?.setData('route_settings_arguments', arguments);
-    }
-
-    await _hub.configureScope((scope) {
-      scope.span ??= _transaction;
-    });
-
-    await _native?.beginNativeFramesCollection();
-  }
-
-  Future<void> _finishTransaction() async {
-    final transaction = _transaction;
-    _transaction = null;
-    if (transaction == null || transaction.finished) {
-      return;
-    }
-    transaction.status ??= SpanStatus.ok();
-    await transaction.finish();
+    await _timeToDisplayTracker?.startTracking(routeName, arguments);
+    completedDisplayTracking?.complete();
```

Comment:
```suggestion
    _completedDisplayTracking?.complete();
```
What is this doing, exactly? If it's just for tests, let's add a comment